tests/system/small/ml/test_preprocessing.py

```diff
@@ -304,6 +304,154 @@ def test_min_max_scaler_normalizes(penguins_df_default_index, new_penguins_df):
+            "kbinsdiscretizer_flipper_length_mm": ["bin_6", "bin_2", "bin_4"],
+        },
+        dtype="string[pyarrow]",
+        index=pd.Index([1633, 1672, 1690], name="tag_number", dtype="Int64"),
+    )
+
+    pd.testing.assert_frame_equal(result, expected, rtol=1e-3)
+
+
+def test_k_bins_discretizer_series_normalizes(
+    penguins_df_default_index, new_penguins_df
+):
+    discretizer = bigframes.ml.preprocessing.KBinsDiscretizer()
+    discretizer.fit(penguins_df_default_index["culmen_length_mm"])
+
+    result = discretizer.transform(
+        penguins_df_default_index["culmen_length_mm"]
+    ).to_pandas()
+    result = discretizer.transform(new_penguins_df).to_pandas()
+
+    # TODO: bug? feature columns seem to be in nondeterministic random order
+    # workaround: sort columns by name. Can't repro it in pantheon, so could
+    # be a bigframes issue...
+    result = result.reindex(sorted(result.columns), axis=1)
+
+    expected = pd.DataFrame(
+        {
+            "kbinsdiscretizer_culmen_length_mm": ["bin_3", "bin_3", "bin_3"],
+        },
+        dtype="string[pyarrow]",
+        index=pd.Index([1633, 1672, 1690], name="tag_number", dtype="Int64"),
+    )
+
+    pd.testing.assert_frame_equal(result, expected, rtol=1e-3)
+
+
+def test_k_bins_discretizer_normalizes(penguins_df_default_index, new_penguins_df):
+    # TODO(http://b/292431644): add a second test that compares output to sklearn.preprocessing.StandardScaler, when BQML's change is in prod.
+    discretizer = bigframes.ml.preprocessing.KBinsDiscretizer()
+    discretizer.fit(
+        penguins_df_default_index[
+            ["culmen_length_mm", "culmen_depth_mm", "flipper_length_mm"]
+        ]
+    )
+
+    result = discretizer.transform(
+        penguins_df_default_index[
+            ["culmen_length_mm", "culmen_depth_mm", "flipper_length_mm"]
+        ]
+    ).to_pandas()
+
+    result = discretizer.transform(new_penguins_df).to_pandas()
+
+    # TODO: bug? feature columns seem to be in nondeterministic random order
+    # workaround: sort columns by name. Can't repro it in pantheon, so could
+    # be a bigframes issue...
+    result = result.reindex(sorted(result.columns), axis=1)
+
+    expected = pd.DataFrame(
+        {
+            "kbinsdiscretizer_culmen_depth_mm": ["bin_5", "bin_4", "bin_4"],
+            "kbinsdiscretizer_culmen_length_mm": ["bin_3", "bin_3", "bin_3"],
+            "kbinsdiscretizer_flipper_length_mm": ["bin_4", "bin_2", "bin_3"],
+        },
+        dtype="string[pyarrow]",
+        index=pd.Index([1633, 1672, 1690], name="tag_number", dtype="Int64"),
+    )
+
+    pd.testing.assert_frame_equal(result, expected, rtol=1e-3)
+
+
+def test_k_bins_discretizer_normalizes_different_params(
+    penguins_df_default_index, new_penguins_df
+):
+    # TODO(http://b/292431644): add a second test that compares output to sklearn.preprocessing.StandardScaler, when BQML's change is in prod.
```

Comment:
same

Reply:
We need to clean this out for all of the preprocessing transformers. b/303153997: Will fix this in a separate ticket.